framework/lib/components/createable-select-dropdown/createable-select-dropdown.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useState } from 'react'
+import { CreatableSelect } from 'chakra-react-select'
+import type { Option } from '@northlight/ui'
+import { PlusSolid } from '@northlight/icons'
+import { Icon } from '@chakra-ui/react'
+import type { CreatableSelectDropdownProps } from './types'
+import { customComponents } from './custom-components'
+
+/**
+ * A dropdown component that allows users to select a value from given options
+ * or create a new one if it doesn't exist.
+ * @see {@link https://northlight.dev/reference/creatable-select-dropdown}
+ *
+ * @example (Example)
+ * ##Select the best artist
+ * (?
+ * () => {
+ *   const [artist, setArtist] = useState(null);
+ *   const options = [
+ *     { label: 'Scooter', value: 'scooter' },
+ *     { label: 'Snoop Doggy Dogg', value: 'snoop-dogg' },
+ *   ];
+ *
+ *   const handleOptionChange = (newOption) => {
+ *     setArtist(newOption);
+ *     // Perform additional actions with the selected or created option
+ *   };
+ *
+ *   return (
+ *     <Box align="left">
+ *       <CreatableSelectDropdown
+ *         standardOptions={options}
+ *         onOptionChange={handleOptionChange}
+ *         width="300px"
+ *       />
+ *       { artist && artist.value !== 'custom-option' && (
+ *         <H3 py={8}>The best artist is: { artist.label }</H3>
+ *       ) }
+ *     </Box>
+ *   );
+ * }
+ * ?)
+ */
+
+export const CreatableSelectDropdown = ({
+  standardOptions,
+  initialPlaceholder = 'Select or create...',
+  addOptionPlaceholder = 'Enter text...',
+  creationOption = {
+    label: 'Add option...',
+    value: 'custom-option',
+  },
+  onOptionChange,
+  width = '100%',
+  variant = 'outline',
+}: CreatableSelectDropdownProps) => {
+  const [ options, setOptions ] = useState(standardOptions)
+  const [ selectedOption, setSelectedOption ] = useState<Option | null>(null)
+  const [ newOptionText, setNewOptionText ] = useState('')
+  const [ newOptionPlaceholder, setNewOptionPlaceholder ] = useState(initialPlaceholder)
+  const [ menuIsOpen, setMenuIsOpen ] = useState(false)
+
+  const handleInputChange = (newValue: string) => {
+    setNewOptionText(newValue)
+  }
+
+  const handleCreateOption = () => {
```

Comment:
yes, it just adds one option to make it selectable. The toLowercase stuff is to prevent duplicate options. The text that the user entered is what is made as an available option (same values for label and value) of the option. This just adds an selectable option, then validation needs to be made on submit that the selected option is valid.

Wouldn't it be better to keep this logic inside the component instead of having the outside component handle it? I think this is internal logic